ee/wallets/pkg/ledger_interface.go

```diff
@@ -193,7 +193,8 @@ func (d DefaultLedger) CreateTransaction(ctx context.Context, ledger string, tra
 				return &transaction.Timestamp.Time
 			}(),
 		},
-		Ledger: ledger,
+		Ledger:         ledger,
+		IdempotencyKey: pointer.For(ik),
```

Comment:
What if the ik is "" ? Should we not add anything in this case ?

Reply:
The ledger will handle the empty string in the same way